CMakeLists.txt

```diff
@@ -372,38 +373,23 @@ if (NANOGUI_EXTRA_RESOURCES)
   list(APPEND resources ${NANOGUI_EXTRA_RESOURCES})
   # Allow parent projects to #include <nanogui/resources.h> without needing to
   # know where it was generated to.
-  list(APPEND NANOGUI_EXTRA_INCS ${CMAKE_CURRENT_BINARY_DIR})
+  list(APPEND NANOGUI_EXTRA_INCS "${NANOGUI_GENERATED_RESOURCE_DIR}/include")
 endif()
 
-# Concatenate resource files into a comma separated string
-string(REGEX REPLACE "([^\\]|^);" "\\1," resources_string "${resources}")
-string(REGEX REPLACE "[\\](.)" "\\1" resources_string "${resources_string}")
-
-# Create command line for running bin2c cmake script
-file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/nanogui")
-set(bin2c_cmdline
-  -DOUTPUT_C=nanogui/resources.cpp
-  -DOUTPUT_H=nanogui/resources.h
-  "-DINPUT_FILES=${resources_string}"
-  -P "${CMAKE_CURRENT_SOURCE_DIR}/resources/bin2c.cmake")
-
-# Run bin2c on resource files
-add_custom_command(
-  OUTPUT nanogui/resources.cpp nanogui/resources.h
-  COMMAND ${CMAKE_COMMAND} ARGS ${bin2c_cmdline}
-  DEPENDS ${resources}
-  COMMENT "Running bin2c"
-  PRE_BUILD VERBATIM)
+# All font resources are now stored in `resources` list.  Execute bin2c at
+# configure time, see top of resources/bin2c.cmake for more information.
+set(BIN2C_OUTPUT_C "${NANOGUI_GENERATED_RESOURCE_DIR}/src/resources.cpp")
```

Comment:
Very minor: since these are now part of the NanoGUI CMake scope, I propose prefixing with ``NANOGUI_``. These variables sometimes show up in visual CMake frontends, so it's nice to immediately see where they come from.

Reply:
WILCO, I'll also prefix `NanoGUI:` in the `message(STATUS`.  The more fonts requested the longer it takes, making nice to say what step is going on ;)

Reply:
So the output looks like this:

```console
-- NanoGUI: generating font resources.
--   Processing: Roboto-Bold.ttf
--   Processing: Roboto-Regular.ttf
--   Processing: RobotoMono-Bold.ttf
--   Processing: RobotoMono-Regular.ttf
--   Processing: entypo.ttf
```

I like having the "Processing" part in there because the step does take a little bit now.  If you would rather not have it just let me know!